src/main/JayZ.ts

```diff
@@ -10,13 +10,19 @@ export type JayZConfig = {
   maxUsesPerDataKey?: number
 }
 
+export type EncryptItemRequest<T, K extends keyof T> = {
+  item: T
+  fieldsToEncrypt: K[]
+}
+
 export class JayZ {
   private keyProvider: DataKeyProvider
   private encryptor: Encryptor = new LibsodiumEncryptor()
 
   private maxUsesPerDataKey: number
-  private timesDataKeyUsed: number = 0
-  private dataKey?: DataKey
+
+  private currentDataKey?: Promise<DataKey>
```

Comment:
Prior to this PR, we'd return a pointer to this data key inside of the `__jayz__metadata` property that we stick on the JSON object we return from `encryptItems`. 

This was the thing that would get zeroed out when we "switched" to a fresh data key once we hit `maxUsesPerDataKey`.